TcgPocket/App/src/pages/settings/modules/PasswordForm.tsx

```diff
@@ -0,0 +1,132 @@
+import { dispatch } from '../../../store/configureStore';
+import { updateUserPassword } from '../../../services/AuthServices';
+import { error, success } from '../../../services/notification';
+
+type PasswordFormProps = {
+  user: UserGetDto;
+  loginUserAfterUpdate: (arg: SignInUserDto) => void;
+};
+
+type PasswordForm = Omit<UserPasswordUpdateDto, 'userName'>;
+
+const initialValues = {
+  currentPassword: '',
+  newPassword: '',
+  newPasswordConfirmation: '',
+} as const;
+
+export function PasswordForm({
+  user,
+  loginUserAfterUpdate,
+}: PasswordFormProps): React.ReactElement {
+  const { classes } = useStyles();
+  const { validatePassword } = useFormValidation();
+
+  const form = useForm({
+    initialValues: initialValues,
+    validateInputOnBlur: true,
+    validate: {
+      currentPassword: (value) => (value === '' ? 'Cannot be Empty' : null),
+      newPassword: (value, values) =>
+        value === values.currentPassword
+          ? 'Must be different than current password.'
+          : validatePassword(value)
+          ? 'Invalid Password'
+          : null,
+      newPasswordConfirmation: (value, values) =>
+        value !== values.newPassword ? 'Passwords do not match' : null,
+    },
+  });
+
+  const handleSubmit = async (values: PasswordForm) => {
+    const userWithUpdatedPassword: UserPasswordUpdateDto = {
+      userName: user.userName,
+      currentPassword: values.currentPassword,
+      newPassword: values.newPassword,
+      newPasswordConfirmation: values.newPasswordConfirmation,
+    };
+
+    const { payload } = await dispatch(
+      updateUserPassword(userWithUpdatedPassword)
+    );
+
+    if (!payload) {
+      return;
+    } else if (payload.hasErrors) {
+      payload.errors.forEach((err) => error(err.message));
+    } else {
+      success('Password Updated');
+      form.reset();
+
+      loginUserAfterUpdate({
+        userName: user.userName,
+        password: values.newPassword,
+      });
+    }
+  };
+
+  const handleReset = () => {
+    form.reset();
+  };
+
+  const determineDisabled = !form.isValid();
+
+  return (
+    <form onSubmit={form.onSubmit(handleSubmit)} onReset={handleReset}>
+      <header> Password </header>
+
+      <PasswordInput
+        className={classes.passwordInput}
+        label="Current Password"
+        {...form.getInputProps('currentPassword')}
+      />
+
+      <div className={classes.newPasswordContainer}>
+        <PasswordInput
+          className={classes.passwordInput}
+          label="New Password"
+          {...form.getInputProps('newPassword')}
+        />
+
+        <PasswordInput
+          className={classes.passwordInput}
+          label="Confirm Password"
+          {...form.getInputProps('newPasswordConfirmation')}
+        />
+      </div>
+
+      <Flex gap={8} justify={'flex-end'} sx={{ paddingTop: '8px' }}>
+        <SecondaryButton type="reset"> Cancel </SecondaryButton>
```

Comment:
A cancel button shouldn't be clickable if it's not going to do anything, so it should be disabled until the user touches the form. i.e. the form needs to be `dirty` to be clickable

```suggestion
        <SecondaryButton type="reset" disabled={!form.dirty()} > Cancel </SecondaryButton>